src/convertToObject.js

```diff
@@ -12,7 +12,22 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const convertedToObject = {};
+
+  const TempArray = sourceString.split(';').map(item => item.trim());
+
+  const TempArrayFiltred = TempArray.filter(item => item.length !== 0);
+
+  const TempArrayOfArrayes = TempArrayFiltred
+    .map(
+      item => item.split(':').map(itemIn => itemIn.trim())
+    );
```

Comment:
You can use iteration methods in chain instead of using extra variables
```suggestion
  const TempArray = sourceString.
  	split(';')
  	.map(item => item.trim())
  	.filter(item => item.length !== 0)
  	.map(
      item => item.split(':').map(itemIn => itemIn.trim())
    );
```